public_html/js/index.js

```diff
@@ -185,5 +194,57 @@
 				} );
 			}
 		}
+
+		/**
+		 * Sets up the WikiProject selector and adds listener to update the view
+		 */
+		function setupSelect2() {
+			var $select2Input = $( '#wikiproject-selector' );
+			var params = {
+				ajax: {
+					url: 'https://en.wikipedia.org/w/api.php',
+					dataType: 'jsonp',
+					delay: 200,
+					jsonpCallback: $.noop(), // have to provide some function or else Select2 gets angry
+					data: function( search ) {
+						return {
+							action: 'query',
+							list: 'prefixsearch',
+							format: 'json',
+							pssearch: 'Wikipedia:WikiProject ' + ( search.term || '' )
```

Comment:
We should be excluding redirects here, since some redirects that look like WikiProjects aren't, for example:
https://en.wikipedia.org/w/index.php?title=Wikipedia:WikiProject_Animals_in_The_Signpost&redirect=no
https://en.wikipedia.org/w/index.php?title=Wikipedia:WikiProject_Comics_in_The_Signpost&redirect=no